integration/cdi/src/main/java/org/apache/cxf/cdi/JAXRSCdiResourceExtension.java

```diff
@@ -74,16 +74,16 @@
             featureBeans.add((Bean< ? extends Feature >)event.getBean());
         } else if (CdiBusBean.CXF.equals(event.getBean().getName())
                 && Bus.class.isAssignableFrom(event.getBean().getBeanClass())) {
-            busBean = event.getBean();
+            hasBus = true;
         }
     }
 
     public void load(@Observes final AfterDeploymentValidation event, final BeanManager beanManager) {
+        // no need of creational context, it only works for app scoped instances anyway
+        final Bean<?> busBean = beanManager.resolve(beanManager.getBeans(CdiBusBean.CXF));
         bus = (Bus)beanManager.getReference(
```

Comment:
Concretely there is a single reliable way - resolve(getBeans()). That said the cases where this is an issue for providers and resources are very limit today for JAX-RS (but can create some nice feature if CXF embraces - typically reading metadata like @Path from cdi AnnotatatedMethod instead of doing pure reflection).

For features this is a valid point I think.

So concretely what does it mean? That discovered instances through CDI scanning (collect()) is only valid for @Produces or managed bean. Think it is fair to start with that but happy to try to do another PR fixing it more if you think it makes sense.